src/js/express.js

```diff
@@ -0,0 +1,148 @@
+        Object.keys(that.options.components).forEach(function(key) {
+            var component = that[key];
+            if (fluid.hasGrade(component.options, "gpii.express.router")) {
+                component.events.addRoutes.fire();
+                that.model.router.use(that.options.path, component.model.router);
+            }
+            else if (fluid.hasGrade(component.options, "gpii.express.middleware")) {
+                if (component.model.middleware) {
+                    var middlewareToLoad = Array.isArray(component.model.middleware) ? component.model.middleware : [component.model.middleware];
+                    for (var i = 0; i < middlewareToLoad.length; i++) {
+                        var functionName = middlewareToLoad[i];
+                        if (component[functionName]) {
+                            try {
+                                that.model.router.use(component[functionName]);
+                            }
+                            catch (e) {
+                                console.error("Error loading middleware function '" + functionName + "' in module '" + key + "':" + e);
+                            }
+                        }
+                        else {
+                            console.error("Middleware '" + functionName + "' does not exist in module '" + key + "'...");
+                        }
+                    }
+                }
+                else {
+                    console.log("Middleware module has no middleware to load, can't continue...");
+                }
+            }
+            else if (fluid.hasGrade(component.options, "gpii.express.helper")) {
+                var helpers = component.getHelpers();
+                Object.keys(helpers).forEach(function(key){
+                    if (helpersToLoad[key]) {
+                        console.error("Another module has already loaded a helper named '" + key + "'. This instance will not be enabled.");
+                    }
+                    else {
+                        helpersToLoad[key] = helpers[key];
+                    }
+                });
+            }
+        });
+    }
+
+    if (that.options.config.express.views) {
+        var viewRoot = that.options.config.express.views;
+        var handlebarsConfig = {
+            defaultLayout: "main",
+            layoutsDir:    viewRoot + "/layouts/",
+            partialsDir:   viewRoot + "/partials/"
+        };
+
+        if (helpersToLoad) {
+            handlebarsConfig.helpers = helpersToLoad;
+        }
+
+
+        that.model.express.set("views", viewRoot);
+
+        var hbs = exphbs.create(handlebarsConfig);
+        that.model.express.engine("handlebars", hbs.engine);
+        that.model.express.set("view engine", "handlebars");
+    }
+    else {
+        console.error("Cannot initialize template handling without a 'config.express.views' option");
+    }
+};
+
+express.startPrivate = function(that, callback) {
```

Comment:
As discussed in the mail server review, I have refactored to remove callbacks and use events and listeners instead.